pandas/io/parsers.py

```diff
@@ -999,6 +999,8 @@ def _failover_to_python(self):
 
     def read(self, nrows=None):
         if nrows is not None:
+            nrows = _validate_integer('nrows', nrows)
```

Comment:
yes this change should be done in the parser itself. See if you can come up with an example that ONLY used pd.read_csv directly.